test/tools/WebListener/Controllers/LinkController.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.Extensions.Primitives;
+using mvc.Models;
+
+namespace mvc.Controllers
+{
+    public class LinkController : Controller
+    {
+        public JsonResult Index()
+        {
+            if (!Request.Query.TryGetValue("maxlinks", out StringValues maxLinksSV) || !Int32.TryParse(maxLinksSV.FirstOrDefault(), out int maxLinks) || maxLinks < 1)
+            {
+                maxLinks = 3;
+            }
+
+            if (!Request.Query.TryGetValue("linknumber", out StringValues linkNumberSV) || !Int32.TryParse(linkNumberSV.FirstOrDefault(), out int linkNumber) || linkNumber < 1)
+            {
+                linkNumber = 1;
+            }
+
+            string baseUri = Regex.Replace(UriHelper.GetDisplayUrl(Request), "\\?.*", String.Empty);
```

Comment:
So simple? 👍

Reply:
Something wrong with it?

Reply:
No, I expected a url parsing 😄

Reply:
heh. So far as I can tell, the only way to get the request url is `GetDisplayUrl()` which returns a string. I could use UriBuilder on that string.. BUT.. on macOS UriBuilder behaves differently with the query section resulting in invalid urls.. rather than mess with that, the regex works just fine :)